osu.Game/Online/Chat/Channel.cs

```diff
@@ -51,11 +54,18 @@ public void AddLocalEcho(LocalEchoMessage message)
             NewMessagesArrived?.Invoke(new[] { message });
         }
 
+        public bool MessagesLoaded { get; private set; }
+
         public void AddNewMessages(params Message[] messages)
         {
             messages = messages.Except(Messages).ToArray();
 
             Messages.AddRange(messages);
+            MessagesLoaded = true;
+
+            var maxMessageId = messages.Max(m => m.Id);
+            if (maxMessageId > LastMessageId)
+                LastMessageId = maxMessageId;
```

Comment:
messages.Last().Id? It's guaranteed to be >= LastMessageId because it's a sorted list.

Reply:
I don't think this will necessarily be correct, since the last_message_id is coming from the server and may be higher than the local message store (we could be retrieving a backlog of messages).